plugins/outputs/http/http.go

```diff
@@ -236,13 +268,44 @@ func (h *HTTP) writeMetric(reqBody []byte) error {
 
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
 
+		if h.MaxRetries == 0 {
+			// Drop, never retry
+			h.Log.Debugf("%s  Error HTTP status %v. MaxRetries %d so NEVER retry.", 
+						h.URL, resp.StatusCode, h.MaxRetries)
+			return nil
+		}
+
 		// Port from v1.22.0
 		for _, nonRetryableStatusCode := range h.NonRetryableStatusCodes {
 			if resp.StatusCode == nonRetryableStatusCode {
 				h.Log.Errorf("%s Received non-retryable status %v. Metrics are lost.", h.URL, resp.StatusCode)
 
 				if h.MaxRetries > 0 {
-					atomic.StoreUint32(&h.FailCount, 0)
+					atomic.StoreInt32(&h.FailCount, 0)
+				}
+				return nil
+			}
+		}
+
+		// EXTR - Only retry if specific retry list is provided HTTP response code is present in list
+		if len(h.RetryableStatusCodes) > 0 {
+			
+			var retryableCodeFound bool = false
+
+			for _, retryableStatusCode := range h.RetryableStatusCodes {
+				if resp.StatusCode == retryableStatusCode {
+					h.Log.Debugf("%s Received retryable status %v. Metrics will be retried.  h.MaxRetries:%d", 
+								h.URL, resp.StatusCode, h.MaxRetries, )
+					retryableCodeFound = true
+					break
+				}
+			}
+
+			if retryableCodeFound == false {
+				h.Log.Errorf("%s Received status %v not found in retryable code list. Metrics are dropped", h.URL, resp.StatusCode)
+
+				if h.MaxRetries > 0 {
+					atomic.StoreInt32(&h.FailCount, 0)
```

Comment:
Would be good to count the error codes in this failcount category for post-mortem troubleshooting?  It would be good to know how many of each status code we are getting if that is not already counted elsewhere.

Reply:
Sounds like a good idea. Given this is going in 33.2, I'll looking into adding failed response code counters in the next  iteration .